### ISSUE TITLE
ft_timelockgrandaverage error when averaging over subjects where some have interpolated channels

### ISSUE THREAD

**Issue by NirOfir** [fieldtrip/fieldtrip#1814]
**Describe the bug**
A recent change to ft_timelockgrandaverage (PR #1777, lines 260-267) introduced handling of cases where channels were interpolated. The code makes sure that all datasets have the same electrode positions, judged by an elec field. This field is introduced whenever ft_repairchannel is used to interpolate data, but isn't necessarily there in the data before that as far as I can tell. This means that if I use a data which does not come with an elec field in its header (such as gtec's matfiles), and only some subjects have bad channels I want to interpolate, then some of the datasets given to ft_timelockgrandaverage will have an elec field and some will not. Then, at some point in the loop, the code is going to get a dataset that doesn't have the elec field and crash. I see 2 possible issues with the code now:
1. Unless elec fields are introduced for all data structure early enough in the pipeline (ft_read_data? ft_preprocessing?), this issue might pop up for all sorts of data types.
2. Related to the first - The current code only checks if the first dataset has the elec field. If I happen to not interpolate any channels for the first subject, this will be invisible to the code.

**To Reproduce**
I think the issue is quite clear in the code, but I can create test data if needed.

**Expected behavior**
I'm not sure. Is the elec field expected to always be there? If so, I guess the fix could be a pretty big job. If not, maybe there should be a test if all datasets have the field, before checking if they all have the same positions?

**Comment by robertoostenveld** [comment]
> some of the datasets given to ft_timelockgrandaverage will have an elec field and some will not

This is something that I did not consider. Right now the code looks at the first data structure, and if that has an elec field it assumes that all others also have an elec field.  

Could you check whether it is solved with a  change around line 240 when 

    keepelec = keepelec && isequal(varargin{1}.elec, varargin{i}.elec);

becomes

    keepelec = keepelec && isfield(varangian{i}, 'elec') && isequal(varargin{1}.elec, varargin{i}.elec);

?

**Comment by robertoostenveld** [comment]
for reference, `private/append_common.m` does something similar, but solves it differently (as it has cfg.appendsens=yes/no).

**Comment by NirOfir** [comment]
Thanks, it does work. I think the append_common approach is better, but I wrote it a bit more concisely:
```
haselec = cellfun(@(x) isfield(x, 'elec'), varargin(:));
if all(haselec) % check whether positions are different between subjects
```
And the same for grad and opto. This works on my data, and if you think that's a reasonable approach I would be happy to make a PR.

**Comment by robertoostenveld** [comment]
this is closed with #1815
